### PR TITLE
contrib: remove vim syntax

### PR DESCRIPTION
**Pull request by alltilla**
The source moved to https://github.com/syslog-ng/vim-syslog-ng

### CODE REVIEW

**Comment by alltilla** [comment]
@czanik How important is it to have the vim syntax file in the RHEL package? The one stored in this source code has not been updated for a long time. We have moved it to another git repo, which gets updated regularly, can the users use that instead?

**Comment by czanik** [comment]
@alltilla  Tough question. I only realized that it's part of the Fedora / RHEL packaging, when it was updated recently: https://github.com/syslog-ng/syslog-ng/pull/4421 :-)

The problem is that it is part of the base syslog-ng package. I'd be happy to see it removed, and forget about it. However, that would only work if someone starts to package vim support from the new git repo. I do not want to be that person, I'd rather try to decrease the number of packages I maintain.

TL;DR; I want to keep vim support as is, unless someone steps up and starts packaging vim support.

**Comment by alltilla** [comment]
Summoning @smortex :)

Do you know what the packaging of vim syntax files look like nowadays? Is it by RHEL packages or with a different solution? I saw you mention Plug and Pathogen. If the users generally don't use RHEL packages for this, then we can safely remove it from the RHEL packaging and let the users get the files with their chosen tool.

**Comment by smortex** [comment]
> Do you know what the packaging of vim syntax files look like nowadays? Is it by RHEL packages or with a different solution? I saw you mention Plug and Pathogen. If the users generally don't use RHEL packages for this, then we can safely remove it from the RHEL packaging and let the users get the files with their chosen tool.

As far as I am concerned, I think I never installed packages to add some vim syntax support: some packages ship with config and it is great (e.g. vim itself, syslog-ng rpm), but for the rest I use a plugin manager.  I currently use [plug](https://github.com/junegunn/vim-plug), I know that [pathogen](https://github.com/tpope/vim-pathogen) and [vundle](https://github.com/VundleVim/Vundle.vim) exist and probably do the same.  I can't speak for all vim users, but I have been using plug *for years*, and I think I read about it when I wanted to install some plugin that recommended to rely on a plugin manager.  My opinion is that it is a best-practice, and I recommend it to anyone who wants to add plugins to vim.

So my recommendation would be to chose one of these:

 1. **If the syntax is installed by the syslog-ng package** (as done by the rpm currently), I would expect it to always be up-to-date, and also for consistency also expect it to be installed by the deb package.  In short: no action required to use the syntax in vim, 100% automated.
 2. **If the syntax is *not* installed by the syslog-ng package**, it would expect the syntax to be in a dedicated git repo usable by a plugin manager like the repo I created.  In order for it to be perceived as a "source of truth", I would expect that repo to be into the *syslog-ng* organization.  In short: the user is in control and install what they want if they wand and at the version they want, 0% automated.

Both are equally fine to me :shrug:.

**Comment by alltilla** [comment]
Thanks Romain! :)

@czanik Are you okay with the 2nd method? We could move the project to the syslog-ng org, and remove it from the RPM packaging.

**Comment by czanik** [comment]
@alltilla @smortex  Yes, I'm OK with the 2nd method.

**Comment by smortex** [comment]
Cool!  I tried to transfer the repo to:
* to the @syslog-ng organization but it failed with "You don’t have the permission to create public repositories on syslog-ng";
* to @alltilla bot got "alltilla/vim-syslog-ng already exists";
* to @czanik  and it suceeded with " Repository transfer to czanik requested" :tada: 

You should have a notification I guess to accept the transfer and expect you to be authorized to transfer it to the right place :+1:

**Comment by alltilla** [comment]
@smortex Awesome, thank you! :)
@czanik Can you ask around in OI to transfer the czanik/vim-syslog-ng repo to the syslog-ng org? Thanks!

**Comment by czanik** [comment]
I'm part of the syslog-ng org, so it arrived to it's new location: https://github.com/syslog-ng/vim-syslog-ng

Of course it still might need some configuration, as all I did is accepting to my repo and then transferring to syslog-ng.

**Review by MrAnno — APPROVED** [review]

**Comment by czanik** [comment]
@alltilla @smortex One more thing: this should be added to the release notes with instructions how people can download vim syntax, now that it's removed from the package.

**Comment by smortex** [comment]
Quick draft for such a release note:

> vim syntax files where previously installed by the RedHat packages of syslog-ng (but not the Debian ones).  These files where sometime lagging behind, so in order to provide a more up-to-date experience on all platforms, regardless of the installation of the syslog-ng package, the vim syntax files have been moved to a dedicated repository [syslog-ng/vim-syslog-ng](https://github.com/syslog-ng/vim-syslog-ng) that can be used using a plugin manager such as [vim-plug](https://github.com/junegunn/vim-plug), [vim-pathogen](https://github.com/tpope/vim-pathogen) or [vundle](https://github.com/VundleVim/Vundle.vim).

**Comment by alltilla** [comment]
Thanks Romain!

I will create a newsfile entry PR with this.
